### ISSUE TITLE
LineChart not use all space

### ISSUE THREAD

**Issue by ruiaraujolink** [indiespirit/react-native-chart-kit#538]
The LineChart does not use all space available, what can I do?

![Screenshot_2021-07-15-16-41-29-772_host exp exponent (1)](https://user-images.githubusercontent.com/87080444/125818246-dbc54e2c-7162-4128-9cf2-700d5172a18c.jpg)

My code on [snack](https://snack.expo.io/@rui.araujo/chartnotuseallspace)

**Comment by tlow92** [comment]
You used fromZero which sets minimum to 0

**Comment by ruiaraujolink** [comment]
Right, but I want to remove what is behind the orange mark, sorry if the drawing is not clear enough

**Comment by abhiburk** [comment]
Did you solved the issue ? I am facing the same issue

**Comment by ruiaraujolink** [comment]
> Did you solved the issue ? I am facing the same issue

No, I'm still facing this issue.

**Comment by luco** [comment]
Same here
![image](https://user-images.githubusercontent.com/246373/129111892-1004e931-cfbb-43b7-9946-fde016f966fe.png)

**Comment by anwersolangi** [comment]
Anyone who solved the issue?

**Comment by ruiaraujolink** [comment]
> Anyone who solved the issue?

I don't

**Comment by phoenixbeats01** [comment]
Having the same issue.

**Comment by lucidprojects** [comment]
wow @tlow92 that is a useful package!  patch-package installed.   Patched chart-kit package attached.
[react-native-chart-kit.zip](https://github.com/indiespirit/react-native-chart-kit/files/7464017/react-native-chart-kit.zip)
 
cheers ✌️

**Comment by leidto** [comment]
It's not working with area chart.

**Comment by mynameisguy** [comment]
Much simpler approach if you have at least 2 data points is to change the divider to be length - 1 instead of length.
For example, change:
`Math.floor(paddingRight + (i * (width - paddingRight)) / dataset.data.length)`
to
`Math.floor(paddingRight + (i * (width - paddingRight)) / (dataset.data.length-1))`

**Comment by ZaharGusyatin** [comment]
any updates?

**Comment by ZaharGusyatin** [comment]
okay, if u can and want to fork, u can find this
Math.floor(paddingRight + (i * (width - paddingRight)) / dataset.data.length)
and change to
Math.floor(paddingRight + (i * (width - paddingRight)) / (dataset.data.length-1))
if u don`t want to fork, u can use this:
  <LineChart
          data={{
            datasets: [{
              data: data
            }]
          }}
          width={widthW + widthW / (data.length - 1)}
when widthW is Dimensions.get('window').width

**Comment by lzagar-itm** [comment]
Are there any updates?

**Comment by karonator** [comment]
> Are there any updates?

You can use version from my fork, I fixed this issue there... https://github.com/karonator/react-native-chart-kit

**Comment by mbalazy** [comment]
Are there any updates?

**Comment by Girish-Official-CBT** [comment]
any updates ? still facing issue.

**Comment by mauricebutts** [comment]
Yea would like a fix here too, is this repo active anymore?

**Comment by renatomserra** [comment]
facing this issue too, i think this repo is not maintained anymore

**Comment by j-peace** [comment]
I Know that is not the best way:

Use Dimensions:


```
Style={{
          paddingRight: Dimensions.get('window').width * 0.04, // (change to the percentage that fit better for you)
          paddingLeft: Dimensions.get('window').width * 0.16
}}
```

**Comment by haysjanete** [comment]
I have found a solution to my issue by wrapping my chart component into a ScrollView rather then just a View and that helped my chart go all the way.

**Comment by rahul-rayo** [comment]
> wow @tlow92 that is a useful package! patch-package installed. Patched chart-kit package attached. [react-native-chart-kit.zip](https://github.com/indiespirit/react-native-chart-kit/files/7464017/react-native-chart-kit.zip)
> 
> cheers ✌️


> https://github.com/xanderdeseyn/react-native-responsive-linechart

In android working fine but in ios not wokring

**Comment by Haze787** [comment]
I tried the solution provided but it doesn't work correctly. shifting the lines causes there to be empty space. I cant figure out how to remove the empty space. 

<img width="273" height="293" alt="Image" src="https://github.com/user-attachments/assets/40d6e936-ce00-4676-9b1c-5e4083dc5b40" />
